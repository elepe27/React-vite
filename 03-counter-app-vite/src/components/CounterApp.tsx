import { props } from "../Interfaces/propsCounterApp";
import PropTypes from 'prop-types'
import { useState } from 'react';




export const CounterApp = ({value} :props) => {
    const [counter, setcounter] = useState(value);

    const handleAdd = ()=> setcounter(counter+1);
    const hanldeSubstract = ()=> setcounter(counter -1);
    const handleReset = ()=> setcounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <hr/>
            <button 
                onClick={ handleAdd } > +1</button>

            <button 
                aria-label="btn-reset"
                onClick={ handleReset } >Reset</button>
            <button 
                onClick={ hanldeSubstract } > -1</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}



