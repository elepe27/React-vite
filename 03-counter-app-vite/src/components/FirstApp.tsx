import { props } from "../Interfaces/propsSaludoApp";
import PropTypes from 'prop-types';

/**Componente principal
 * @param arg:object
 */
export const FirstApp = ( {title, subtitle, edad , name} : props ) => {

    return (
        <>
            <h1 data-testid='test-title'>{ title }</h1>
            {/* <code> { JSON.stringify( nombre, 0, null, 4 ) } </code> */}
            <h2>{ subtitle }</h2>
            <h2>{ edad }</h2>
            <h2>{ name }</h2>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    edad: 28,
    name: 'Esteban Lepe',
    title: 'No hay Titulo',
}