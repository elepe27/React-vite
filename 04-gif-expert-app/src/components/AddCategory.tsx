import { useState } from 'react';
import { IMyProps } from '../Interface/AddCategory/IAddCategoryProps';

export const AddCategory:React.FC<IMyProps> = ( { onNewCategory }:IMyProps ) => {

    /**Estado inicial de nuestro componente */
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target: { value } }: any) => setInputValue(value);

    const onSubmit = ( e :any )=>{
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;
            
        onNewCategory(inputValue);
        setInputValue('');
       
    }

    return (
        <form onSubmit={ onSubmit }>
            <input
                type='text'
                placeholder='Buscar gifs'
                value={ inputValue }
                onChange={ onInputChange }
                
            />
        </form>
    )
}