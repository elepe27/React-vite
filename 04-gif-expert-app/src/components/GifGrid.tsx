import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }: any) => {


    const { images, isLoading } = useFetchGifs(category);

    return (

        <>

            {
                isLoading
                    ?
                    <p>Loading </p>

                    :
                    <>

                        <h3> {category} </h3>
                        <div className='card-grid'>
                            {images.map((img: any) => <GifGridItem {...img} key={img.id} />)}
                        </div>

                    </>
            }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.any.isRequired
}