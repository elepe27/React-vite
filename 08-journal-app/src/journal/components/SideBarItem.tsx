import { useMemo } from 'react';

import { ListItem, ListItemButton, ListItemIcon, Grid, ListItemText } from '@mui/material';
import { TurnedInNot } from '@mui/icons-material';

import { setActiveNote } from '../../store/journal';
import { useDispatch } from 'react-redux';

export const SideBarItem = ( { title='', body, id , date, imageUrls= [] }:any ) => {
    

    const newTitle = useMemo( ()=> {
        return title.length > 17
            ? title.substring(0,17) + '...'
            : title;
    },[ title ] );

    const dispatch:any = useDispatch();

    const onActiveNote = ()=> {

        const obj = {id, date, title, body, imageUrls};

        dispatch(setActiveNote(obj));     
    }
    return (
        <ListItem key={id} disablePadding>
            <ListItemButton onClick={ onActiveNote }>
                <ListItemIcon>
                    <TurnedInNot />
                </ListItemIcon>
                <Grid container>
                    <ListItemText primary={ newTitle } />
                    <ListItemText secondary={ body }/>
                </Grid>
            </ListItemButton>
        </ListItem>
    )
}

