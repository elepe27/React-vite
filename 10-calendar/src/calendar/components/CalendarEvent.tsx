

export const CalendarEvent = ({ event }: any) => {

    const { title, user }: any = event;


    return (
        <>
            <strong> {title}</strong>
            <span> - {user.name} </span>
        </>
    )
}
