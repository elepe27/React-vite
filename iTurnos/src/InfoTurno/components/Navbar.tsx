import { MenuOutlined } from "@mui/icons-material"
import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material"


export const Navbar = ( {drawerWidth}:any ) => {
  return (
    <AppBar
            position="fixed"
            sx={{
                width: {
                    sm: `calc(100% - ${drawerWidth}px)`,
                    ml: { sm: `${drawerWidth}px` }
                }
            }}>
            <Toolbar>
                <IconButton
                    color='inherit'
                    edge='start'
                    sx={{ mr: 2, display: { sm: 'none' } }}
                >
                    <MenuOutlined />
                </IconButton>
                <Grid container direction='row' justifyContent='space-between' alignItems='center'>
                    <img src="/Turnos.png" width='10%' style={{ marginBottom: '20px' }} />
                </Grid>
            </Toolbar>
        </AppBar>
  )
}
