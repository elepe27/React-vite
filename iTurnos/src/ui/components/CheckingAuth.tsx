import { CircularProgress, Grid } from "@mui/material"

export const CheckingAuth = () => {
    return (


        <Grid
            container
            spacing={0}
            direction="column"
            alignItems="center"
            justifyContent="center"
            sx={{
                width: { sd: 450 },
                minHeight: '100vh',
                backgroundColor: 'primary.main',
                padding: 4
            }}
        >
            <Grid container
                direction='row'
                justifyContent='center'
            >

                <CircularProgress color='warning' />


            </Grid>
        </Grid>




    )
}
