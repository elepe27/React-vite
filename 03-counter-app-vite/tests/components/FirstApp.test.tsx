
import { render } from '@testing-library/react';
import { FirstApp } from '../../src/components/FirstApp';


describe('Pruebas en <First />', () => {

    test('debe hacer match con el snapshot', () => { 
        
        let title = 'Hola soy goku'
        const {container} = render( <FirstApp title={title} subtitle='Goku'/>);


        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el titulo en un h1', () => { 
        
        let title = 'Hola soy goku'
        let subtitle = 'soy un subtitulos'
        const { container, getByText, getByTestId, getAllByText } = render( 
            <FirstApp title={title} 
                      subtitle={subtitle}
                      />);
        expect ( getByText(title) ).toBeTruthy();
        
        // const h1 = container.querySelector('h1');
        // expect(h1?.innerHTML).toContain( title );
        
        expect ( getByTestId('test-title').innerHTML ).toBe(title);

    });

   
    
    
 });