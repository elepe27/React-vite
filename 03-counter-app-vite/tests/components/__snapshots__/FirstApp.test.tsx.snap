// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <First /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <h2>
    Goku
  </h2>
  <h2>
    28
  </h2>
  <h2>
    Esteban Lepe
  </h2>
</div>
`;
